qutip/random_objects.py

```diff
@@ -322,10 +322,10 @@ def rand_ket(N=0, density=1, dims=None, seed=None):
     """
     if seed is not None:
         np.random.seed(seed=seed)
-    if N and dims:
+    if N is not None and dims:
         _check_dims(dims, N, 1)
     elif dims:
-        N = prod(dims[0])
+        N = np.prod(dims[0])
         _check_dims(dims, N, 1)
     else:
         dims = [[N],[1]]
```

Comment:
If `dims` and `N` are both `None` this will set `dims` to `[[None], [1]]` which probably is not what we want. I suggest we raise a `ValueError` instead and add a test for this case.

Reply:
Thanks for pointing this out. I should have caught it before !